satpy/tests/writer_tests/test_cf.py

```diff
@@ -960,6 +960,57 @@ def test_area2lonlat(self):
         self.assertDictContainsSubset({'name': 'longitude', 'standard_name': 'longitude', 'units': 'degrees_east'},
                                       lon.attrs)
 
+    def test_update_encoding(self):
+        import xarray as xr
+        from satpy.writers.cf_writer import CFWriter
+
+        # Without time dimension
+        ds = xr.Dataset({'foo': (('y', 'x'), [[1, 2], [3, 4]]),
+                         'bar': (('y', 'x'), [[3, 4], [5, 6]])},
+                        coords={'y': [1, 2],
+                                'x': [3, 4],
+                                'lon': (('y', 'x'), [[7, 8], [9, 10]])})
+        ds = ds.chunk(2)
+        kwargs = {'encoding': {'bar': {'chunksizes': (1, 1)}},
+                  'other': 'kwargs'}
+        enc, other_kwargs = CFWriter.update_encoding(ds, kwargs)
+        self.assertDictEqual(enc, {'y': {'_FillValue': None},
+                                   'x': {'_FillValue': None},
+                                   'lon': {'chunksizes': (2, 2)},
+                                   'foo': {'chunksizes': (2, 2)},
+                                   'bar': {'chunksizes': (1, 1)}})
+        self.assertDictEqual(other_kwargs, {'other': 'kwargs'})
+
+        # Chunksize may not exceed shape
+        ds = ds.chunk(8)
+        kwargs = {'encoding': {}, 'other': 'kwargs'}
+        enc, other_kwargs = CFWriter.update_encoding(ds, kwargs)
+        self.assertDictEqual(enc, {'y': {'_FillValue': None},
+                                   'x': {'_FillValue': None},
+                                   'lon': {'chunksizes': (2, 2)},
+                                   'foo': {'chunksizes': (2, 2)},
+                                   'bar': {'chunksizes': (2, 2)}})
+
+        # With time dimension
+        ds = ds.expand_dims({'time': [datetime(2009, 7, 1, 12, 15)]})
+        kwargs = {'encoding': {'bar': {'chunksizes': (1, 1, 1)}},
+                  'other': 'kwargs'}
+        enc, other_kwargs = CFWriter.update_encoding(ds, kwargs)
+        self.assertDictEqual(enc, {'y': {'_FillValue': None},
+                                   'x': {'_FillValue': None},
+                                   'lon': {'chunksizes': (2, 2)},
+                                   'foo': {'chunksizes': (1, 2, 2)},
+                                   'bar': {'chunksizes': (1, 1, 1)},
+                                   'time': {'_FillValue': None,
+                                            'calendar': 'proleptic_gregorian',
+                                            'units': 'days since 2009-07-01 12:15:00'},
+                                   'time_bnds': {'_FillValue': None,
+                                                 'calendar': 'proleptic_gregorian',
+                                                 'units': 'days since 2009-07-01 12:15:00'}})
+
+        # User-defined encoding may not be altered
+        self.assertDictEqual(kwargs['encoding'], {'bar': {'chunksizes': (1, 1, 1)}})
+
```

Comment:
This can maybe be made into an own testcase and then split in 2 methods?

Reply:
Good idea!